docs/docs/resources/pipeline.yaml

```diff
@@ -0,0 +1,334 @@
+# `pipeline.yaml` contains the configurations for all components to be deployed
```

Comment:
This whole file is very repetitive. Please make it clear that all configs of super classes can also be set on sub-classes so that redundancy is reduced

Reply:
I broke it up into separate files, I would prefer keep each of them as detailed as possible. It would be easier to generate and to also use as a user that has no understanding of `KPOps`. The inheritance should be explained on the page on `defaults.yaml` in my opinion. 

WDYT?

Reply:
I can understand both sides. The problem is that with repeating everything, maintenance effort also increases when adding new field because we have to update the docs everywhere. So unless behavior is different from super class I still tend to not repeat

Reply:
@raminqaf @disrupted what do you think?